sky/backends/cloud_vm_ray_backend.py

```diff
@@ -3001,6 +3086,16 @@ def teardown_no_lock(self,
         config = common_utils.read_yaml(handle.cluster_yaml)
         cluster_name = handle.cluster_name
         use_tpu_vm = config['provider'].get('_has_tpus', False)
+
+        if isinstance(cloud, clouds.AWS):
+            region = config['provider']['region']
+            cloud_vm_backend.teardown_cluster(repr(cloud), region, cluster_name,
+                                              terminate)
+            if post_teardown_cleanup:
+                return self.post_teardown_cleanup(handle, terminate, purge)
+            else:
+                return True
```

Comment:
Nice! We are finally getting rid of the special case handling for the termination.

Reply:
Do we want to remove L3116-L3129?

Reply:
yes, but not in this PR

Reply:
Oh, it seems the later code is unreachable. Maybe worth removing it in this PR to reduce the future confusion?

Reply:
I am ok to remove it, but that means (1) we will enable this PR by default later (2) we should also delete the AWS node provider then. If you think that's fine, I will remove all aws unreachable code.

Reply:
for me I just think it will be more clear to perform a giant cleanup later, that generally helps reviewing